sleap/nn/inference.py

```diff
@@ -4205,6 +4074,245 @@ def export_model(
             )
 
 
+# TODO(LM): Does not support centered instance which needs to run centroid beforehand.
+@attr.s(auto_attribs=True)
+class VisualPredictor:
+    """Wrapper class for generating the visual output of model.
+
+    Initialize this class by using `VisualPredictor.from_model_paths` classmethod.
+
+    Attributes:
+        predictor: The subclass of `Predictor` used to generate predictions.
+        config: The `TrainingJobConfig` containing the metadata for the trained model.
+        model: A `Model` instance created from the trained model.
+        pipeline: A `Pipeline` that loads the data and batches input data.
+
+    Note: The `VisualPredictor` class does not inherit from `Predictor` and is instead
+        designed to wrap subclasses of the `Predictor` class, using the __getattr__
+        method to grab attributes.
+
+    """
+
+    predictor: Union[
+        SingleInstancePredictor,
+        TopDownPredictor,
+        BottomUpPredictor,
+        TopDownMultiClassPredictor,
+        BottomUpMultiClassPredictor,
+    ]
+    config: TrainingJobConfig
+    model: Model
+    pipeline: Optional[Pipeline] = attr.ib(default=None, init=False)
+
+    def __getattr__(self, attr):
+        """Retrieve attributes and methods from `predictor` if not in `self`."""
+        return getattr(self.predictor, attr)
+
+    @classmethod
+    def get_supported_predictors(
```

Comment:
Until we get the centered instance `"confmap_model"` supported, this would be more accurately named
```suggestion
    def get_expected_predictors(
```